app/dot/cpp/repository/repository/BaseRepository.java

```diff
@@ -73,14 +100,27 @@ public List<T> listWithFilterPaginated(Filter filter, int pageSize, int pageNum,
     }
   }
 
-  @NotNull
-  protected static FindOptions getSortOptions(Sort[] sortBy) {
-    return new FindOptions().sort(sortBy);
+  public List<T> listHistory(String trackingId) {
+    final var historyEntities = new ArrayList<T>();
+
+    getHistoryCollection()
+        .find(eq("trackingId", trackingId))
+        .sort(descending("modifiedAt"))
+        .forEach(historyEntities::add);
+
+    return historyEntities;
   }
 
+  // Morphia sets the Mongo codecs based on POJOs, if the Mongo client is used directly then the
+  // codecs have to be set separately, so Morphia is useful in this sense
   @NotNull
-  protected static FindOptions getOptions(int pageSize, int pageNum) {
-    return new FindOptions().skip(pageNum * pageSize).limit(pageSize);
+  private MongoCollection<T> getHistoryCollection() {
+    // todo check for existence?
```

Comment:
todo or not to do

Reply:
This is actually up for discussion, this method can either throw an error if the history collection was not created for this entity or it can simply create the collection when an element is first inserted but then it would have no validation schema. Then again, if something is about to get inserted in the history collection that means it passed the validation schema of its parent collection.